product_analytic_sale/models/sale_order_line.py

```diff
@@ -0,0 +1,46 @@
+# Copyright 2022 Akretion France (http://www.akretion.com/)
+# @author: Alexis de Lattre <alexis.delattre@akretion.com>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import models
+
+
+class SaleOrderLine(models.Model):
+    _inherit = "sale.order.line"
+
+    def _prepare_invoice_line(self, **optional_values):
+        """
+        If the analytic distribution is not yet set on the sale order line,
+        check on the product level if there is one and transmit it to
+        the invoice line.
+        """
+        vals = super()._prepare_invoice_line(**optional_values)
+        if self.product_id and not self.analytic_distribution:
+            ana_account = (
+                self.product_id.product_tmpl_id._get_product_analytic_accounts()[
+                    "income"
+                ]
+            )
+            if ana_account:
+                self.analytic_distribution = {ana_account.id: 100}
+        return vals
```

Comment:
I have tested with standard Odoo:
- if there is an analytic account on product or category, it is added on the invoice, by the `product_analytic` module
- if an analytic distribution is set on the SO line, it has priority over product analytic

So I think this `_prepare_invoice_line` override is not necessary anymore.